tasks_downstream.py

```diff
@@ -0,0 +1,580 @@
+            docker_compose_file_version = orig_docker_config.get("version")
+    else:
+        docker_compose_file_version = "2.4"
+    docker_config = {
+        "version": docker_compose_file_version,
+        "services": {service: {"command": command}},
+    }
+    docker_config_yaml = yaml.dump(docker_config)
+    file.write(docker_config_yaml)
+    file.flush()
+
+
+def _remove_auto_reload(file, orig_file):
+    with open(orig_file, "r") as fd:
+        orig_docker_config = yaml.safe_load(fd.read())
+    odoo_command = orig_docker_config["services"]["odoo"]["command"]
+    new_odoo_command = []
+    for flag in odoo_command:
+        if flag.startswith("--dev"):
+            flag = flag.replace("reload,", "")
+        new_odoo_command.append(flag)
+    _override_docker_command("odoo", new_odoo_command, file, orig_file=orig_file)
+
+
+def _get_cwd_addon(file):
+    cwd = Path(file)
+    manifest_file = False
+    while PROJECT_ROOT < cwd:
+        manifest_file = (cwd / "__manifest__.py").exists() or (
+            cwd / "__openerp__.py"
+        ).exists()
+        if manifest_file:
+            return cwd.stem
+        cwd = cwd.parent
+        if cwd == PROJECT_ROOT:
+            return None
+
+
+@task
+def write_code_workspace_file(c, cw_path=None):
+    """Generate code-workspace file definition.
+
+    Some other tasks will call this one when needed, and since you cannot specify
+    the file name there, if you want a specific one, you should call this task
+    before.
+
+    Most times you just can forget about this task and let it be run automatically
+    whenever needed.
+
+    If you don't define a workspace name, this task will reuse the 1st
+    `doodba.*.code-workspace` file found inside the current directory.
+    If none is found, it will default to `doodba.$(basename $PWD).code-workspace`.
+
+    If you define it manually, remember to use the same prefix and suffix if you
+    want it git-ignored by default.
+    Example: `--cw-path doodba.my-custom-name.code-workspace`
+    """
+    root_name = f"doodba.{PROJECT_ROOT.name}"
+    root_var = "${workspaceRoot:%s}" % root_name
+    if not cw_path:
+        try:
+            cw_path = next(PROJECT_ROOT.glob("doodba.*.code-workspace"))
+        except StopIteration:
+            cw_path = f"{root_name}.code-workspace"
+    if not Path(cw_path).is_absolute():
+        cw_path = PROJECT_ROOT / cw_path
+    cw_config = {}
+    try:
+        with open(cw_path) as cw_fd:
+            cw_config = json.load(cw_fd)
+    except (FileNotFoundError, json.decoder.JSONDecodeError):
+        pass  # Nevermind, we start with a new config
+    # Launch configurations
+    debugpy_configuration = {
+        "name": "Attach Python debugger to running container",
+        "type": "python",
+        "request": "attach",
+        "pathMappings": [{"localRoot": f"{root_var}/odoo", "remoteRoot": "/opt/odoo"}],
+        "port": int(ODOO_VERSION) * 1000 + 899,
```

Comment:
Also, seems like [tests are failing](https://github.com/Tecnativa/doodba-copier-template/pull/168/checks?check_run_id=1373860930#step:13:582) because of this conversion. This could be a workaround:
```suggestion
        "port": int(float(ODOO_VERSION)) * 1000 + 899,
```

Reply:
Thanks, solved in https://github.com/Tecnativa/doodba-copier-template/pull/168/files#diff-9fccd0ad3d50013f36bd06d3ccede328be22fe7d5d1009f445ff8d7f6de0f451R23 instead.